README.md

```diff
@@ -72,6 +72,27 @@ populated if you simply run `go build`. You have to pass in command
 line flags as defined in `Makefile.INCLUDE` (see `${BUILDFLAGS}`) to
 do that.
 
+## Release checklist
+
+To cut a new release of Prometheus, follow this checklist:
+
+1. Consider updating all or some of the vendored dependencies before the
+   release. Make sure that https://github.com/prometheus/client_golang is
+   vendored at a release version.
+1. Create a PR that updates the `CHANGELOG.md` along with the `version/VERSION` file and get it merged.
+1. Run `make tag` to tag the new release and push it to GitHub.
+1. Head to https://github.com/prometheus/prometheus/releases and create a new release for the pushed tag.
+1. Attach the relevant changelog excerpt to the release on GitHub.
+1. Build `linux-amd64`, `linux-386`, and `darwin-amd64` binaries via `make tarball` and attach them to the release.
+1. Update the `stable` branch to point to the new release (to tag the latest release image on https://registry.hub.docker.com/u/prom/prometheus/):
+
+   ```bash
+   git branch -f stable <new-release-tag>
+   git push origin stable
+   ```
+
+1. Announce the release on the mailing list and on Twitter.
```

Comment:
Also, do what's needed in the doc repo: Merge next-release into master, or whatever it is...